mne/report/report.py

```diff
@@ -0,0 +1,3203 @@
+    t = t.substitute(id=id, img=img, caption=caption, tags=tags, title=title,
+                     image_format=image_format, div_klass=div_klass,
+                     img_klass=img_klass, show=show)
+    return t
+
+
+def _html_code_element(*, id, code, language, title, tags):
+    template_path = template_dir / 'code.html'
+    t = Template(template_path.read_text(encoding='utf-8'))
+    t = t.substitute(id=id, code=code, language=language, title=title,
+                     tags=tags)
+    return t
+
+
+def _html_element(*, id, div_klass, html, title, tags):
+    template_path = template_dir / 'html.html'
+    t = Template(template_path.read_text(encoding='utf-8'))
+    t = t.substitute(id=id, div_klass=div_klass, html=html, title=title,
+                     tags=tags)
+    return t
+
+
+@dataclass
+class _ContentElement:
+    name: str
+    dom_id: str
+    tags: Tuple[str]
+    html: str
+
+
+###############################################################################
+# PLOTTING FUNCTIONS
+
+def _fig_to_img(fig, image_format='png', auto_close=True, **kwargs):
+    """Plot figure and create a binary image."""
+    # fig can be ndarray, mpl Figure, Mayavi Figure, or callable that produces
+    # a mpl Figure
+    import matplotlib.pyplot as plt
+    from matplotlib.figure import Figure
+    if isinstance(fig, np.ndarray):
+        fig = _ndarray_to_fig(fig)
+    elif callable(fig):
+        if auto_close:
+            plt.close('all')
+        fig = fig(**kwargs)
+    elif not isinstance(fig, Figure):
+        from ..viz.backends.renderer import backend, MNE_3D_BACKEND_TESTING
+        backend._check_3d_figure(figure=fig)
+        if not MNE_3D_BACKEND_TESTING:
+            img = backend._take_3d_screenshot(figure=fig)
+        else:  # Testing mode
+            img = np.zeros((2, 2, 3))
+
+        if auto_close:
+            backend._close_3d_figure(figure=fig)
+        fig = _ndarray_to_fig(img)
+
+    output = BytesIO()
+    logger.debug('Saving figure %s with dpi %s'
+                 % (fig.get_size_inches(), fig.get_dpi()))
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter('ignore')  # incompatible axes
+        fig.savefig(output, format=image_format, dpi=fig.get_dpi(),
+                    bbox_to_inches='tight')
+    plt.close(fig)
+    output = output.getvalue()
+    return (output.decode('utf-8') if image_format == 'svg' else
+            base64.b64encode(output).decode('ascii'))
+
+
+def _get_mri_contour_figs(*, sl, n_jobs, mri_fname, surfaces,
+                          orientation, src, show, show_orientation, width,
+                          slices_as_figures):
+    import matplotlib.pyplot as plt
+
+    # Matplotlib <3.2 doesn't work nicely with process-based parallelization
+    from matplotlib import __version__ as MPL_VERSION
+    from pkg_resources import parse_version
+    if parse_version(MPL_VERSION) >= parse_version('3.2'):
```

Comment:
elsewhere we mostly use `from distutils.version import LooseVersion` to do comparisons like this.  Is this approach better / more modern?

Reply:
`distutils` is [being deprecated](https://www.python.org/dev/peps/pep-0632). +1 for adding a `mne.fixes._version_greater(a, b, comp='>=')` or something that uses `pkg_resources` if available and `LooseVersion` if not (`comp` could be `'>='` or `'>'`)

Reply:
Great idea @larsoner, will give it a shot. The deprecation was actually the reason why I picked this approach, yes.

Reply:
@larsoner Please take a look at f29eced and let me know what you think! :)

Reply:
@larsoner FYI I made a followup commit to add some fixes, b6f4f5c